main/src/library/Order.flix

```diff
@@ -237,21 +237,29 @@ instance Order[Float64] {
 
 instance Order[BigDecimal] {
 
-    pub override def less(x: BigDecimal, y: BigDecimal): Bool = $BIGDECIMAL_LT$(x, y)
+    pub override def less(x: BigDecimal, y: BigDecimal): Bool =
+        import java.math.BigDecimal.compareTo(BigDecimal): Int32 \ {};
+        (x `compareTo` y) < 0
 
-    pub override def lessEqual(x: BigDecimal, y: BigDecimal): Bool = $BIGDECIMAL_LE$(x, y)
+    pub override def lessEqual(x: BigDecimal, y: BigDecimal): Bool =
+        import java.math.BigDecimal.compareTo(BigDecimal): Int32 \ {};
+        (x `compareTo` y) <= 0
 
-    pub override def greater(x: BigDecimal, y: BigDecimal): Bool = $BIGDECIMAL_GT$(x, y)
+    pub override def greater(x: BigDecimal, y: BigDecimal): Bool =
+        import java.math.BigDecimal.compareTo(BigDecimal): Int32 \ {};
+        (x `compareTo` y) > 0
 
-    pub override def greaterEqual(x: BigDecimal, y: BigDecimal): Bool = $BIGDECIMAL_GE$(x, y)
+    pub override def greaterEqual(x: BigDecimal, y: BigDecimal): Bool =
+        import java.math.BigDecimal.compareTo(BigDecimal): Int32 \ {};
+        (x `compareTo` y) >= 0
 
     pub def compare(x: BigDecimal, y: BigDecimal): Comparison =
-        if ($BIGDECIMAL_LT$(x, y))
-            Comparison.LessThan
-        else if ($BIGDECIMAL_GT$(x, y))
-            Comparison.GreaterThan
-        else
-            Comparison.EqualTo
+        import java.math.BigDecimal.compareTo(BigDecimal): Int32 \ {};
+        match x `compareTo` y {
```

Comment:
@mlutze How can Flix know this is exhaustive? :O

Reply:
It doesn't. This should not have been merged. See #6026.